### PR TITLE
Setting up the project

### PR DESCRIPTION
**Pull request by MichaHllx**
Creation of a batch script (.bat) that runs under Windows using the command prompt (not for PowerShell, possible improvement).
Creation of a bash script (.sh) that runs on Linux and macOS. This script works with the zsh shell and the bash shell.

Advantages:
- The user has virtually nothing to do thanks to the script.
- The readme is easier to understand and more compact.
- If an error is reported, it's easier to understand what the problem is because all users will do exactly the same thing (i.e. run the script). So it's easier to understand where the problem is coming from.
- The dependencies are centralised in one place.

Possible improvements to my contribution:
- Develop a script that works with the new Windows PowerShell.
- Set up a tutorial to install the correct version of python if necessary (for beginners).
- Find a way to solve the problem of the single dependency that requires a very restrictive version of python.

In conclusion, I think and hope that this contribution will solve some of the problems that users had. In any case, I think it will help users who are less at ease with the domain. And I also hope that it will make maintenance work easier afterwards for the developer.

### CODE REVIEW

File: PROJECT-SETUP.sh
Comment (nickpadd) [inline PROJECT-SETUP.sh:1]: I believe having the user independently download the `.sh` file to clone the repository and install dependencies might introduce some confusion. Here’s why:

1. **For users without Git installed**: The script would fail, as cloning relies on Git being present.
2. **For experienced users**: It’s more intuitive to clone the repository themselves and then run the script from within the project directory. Running the script as is would result in the repository being cloned twice.

### Suggested Changes
1. **Simplify the Workflow**: 
   - Assume users will first clone the repository manually (we should provide clear instructions for it). Then, the script can focus solely on setting up the environment and dependencies within the cloned project directory.

2. **Consistent Naming**:
   - To maintain consistency, I suggest adopting the `snake_case` naming convention for directories and files.
   - Scripts intended for user interaction could include the `run` keyword, making their purpose clearer (e.g., `run_setup.sh`).

File: PROJECT-SETUP.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# Display the platform machine
+platform_machine=$($PYTHON -c 'import platform; print(platform.machine())')
+echo -e "Platform machine: $platform_machine\n"
+
+# Clone the repository in the current directory
+git clone https://github.com/nickpadd/EuropeanFootballLeaguePredictor
+
+# Browse the project directory
+cd EuropeanFootballLeaguePredictor || exit
```
Comment (nickpadd) [inline PROJECT-SETUP.sh:44]:
```suggestion
# Remove these lines entirely; repository cloning should be a manual user step.
```

File: PROJECT-SETUP.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Create a virtual environment
+$PYTHON -m venv EFLPvenv
```
Comment (nickpadd) [inline PROJECT-SETUP.sh:47]:
```suggestion
$PYTHON -m venv EFLPvenv
$PYTHON -m venv eflp_venv
```

File: PROJECT-SETUP.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+# Activate the virtual environment
+source EFLPvenv/bin/activate
```
Comment (nickpadd) [inline PROJECT-SETUP.sh:50]:
```suggestion
source eflp_venv/bin/activate
```

File: PROJECT-SETUP.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+# Install dependencies for the project using the pip associated with the virtual environment
+EFLPvenv/bin/pip install -r requirements.txt
```
Comment (nickpadd) [inline PROJECT-SETUP.sh:53]:
```suggestion
eflp_venv/bin/pip install -r requirements.txt
```

File: README.md [hunk 5]
```diff
@@ -14,48 +14,35 @@ The [EuropeanFootballLeaguePredictor Github Page](https://nickpadd.github.io/Eur
...
+2. Open the terminal and navigate to the folder where the 'PROJECT-SETUP.sh' file is located.
+3. Run the following command in the terminal:
+- macOS/Linux users
+```bash
+bash PROJECT-SETUP.sh
+```
+- Windows users
+```bash
+PROJECT-SETUP.bat
+```
```
Comment (nickpadd) [inline README.md:30]: If the `.sh` and `.bat` scripts are modified to exclude the cloning procedure, these lines should explicitly specify the steps for cloning the repository.

File: README.md [hunk 6]
```diff
@@ -14,48 +14,35 @@ The [EuropeanFootballLeaguePredictor Github Page](https://nickpadd.github.io/Eur
...
-    ```bash
-    poetry install
-    ```
+1. Download the file *PROJECT-SETUP.sh* (**or the '.bat' for Windows users**) from the repository and place it in the
+folder where you want to install the project.
+2. Open the terminal and navigate to the folder where the 'PROJECT-SETUP.sh' file is located.
+3. Run the following command in the terminal:
+- macOS/Linux users
+```bash
+bash PROJECT-SETUP.sh
```
Comment (nickpadd) [inline README.md:25]:
```suggestion
bash run_project_setup.sh
```

File: README.md [hunk 7]
```diff
@@ -14,48 +14,35 @@ The [EuropeanFootballLeaguePredictor Github Page](https://nickpadd.github.io/Eur
...
+```
+- Windows users
+```bash
+PROJECT-SETUP.bat
```
Comment (nickpadd) [inline README.md:29]:
```suggestion
run_project_setup.bat
```

File: README.md [hunk 8]
```diff
@@ -14,48 +14,35 @@ The [EuropeanFootballLeaguePredictor Github Page](https://nickpadd.github.io/Eur
...
-    - Install requirements:
-    ```bash
-    pip install -r requirements.txt
-    ```
+### (Already) installed the project? Let's get started!
+1. Open the terminal and navigate to the project folder *EuropeanFootballLeaguePredictor*.
+2. Run the following command in the terminal:
+- macOS/Linux users
+```bash
+source EFLPvenv/bin/activate
```
Comment (nickpadd) [inline README.md:40]:
```suggestion
source eflp_venv/bin/activate
```

File: README.md [hunk 9]
```diff
@@ -14,48 +14,35 @@ The [EuropeanFootballLeaguePredictor Github Page](https://nickpadd.github.io/Eur
...
+```
+- Windows users
+```bash
+EFLPvenv\Scripts\activate
```
Comment (nickpadd) [inline README.md:44]:
```suggestion
eflp_venv\Scripts\activate
```

File: PROJECT-SETUP.bat
Comment (nickpadd) [inline PROJECT-SETUP.bat:1]: I suggest the same changes as in the ```.sh``` script, as well as changing the name to ```run_project_setup.bat``` for consistency!

**Review by nickpadd — CHANGES_REQUESTED** [review]
Great work! I have left a few comments, primarily regarding naming convention changes and workflow improvements. Specifically:

1. **Naming Conventions**: I suggest using `snake_case` for directory and script names to maintain consistency and follow common conventions. Additionally, including the `run` keyword in user-facing scripts (e.g., `run_setup.sh`) makes their purpose clearer.

2. **Cloning Workflow**: If the `.sh` and `.bat` scripts are modified to exclude repository cloning, ensure the documentation or script comments provide clear instructions for users to manually clone the repository before running the setup scripts. This avoids confusion and unnecessary duplication.

Once these changes are addressed, I’ll be happy to approve and merge.
